debezium-core/src/main/java/io/debezium/jdbc/JdbcValueConverters.java

```diff
@@ -889,7 +889,7 @@ else if (data instanceof Boolean) {
                 r.deliver(NumberConversions.getInteger((Boolean) data));
             }
             else if (data instanceof String) {
-                r.deliver(Integer.valueOf((String) data));
+                r.deliver(Math.round(Float.parseFloat((String) data)));
```

Comment:
As a second thought, given that both MySQL and SQL Server store these decimal defaults as strings, the right approach might be to not parse them by the connector at all. AFAIK, the connector doesn't use them and at most will pass them down the line to the consumers which would have to parse the.